### PR TITLE
Issue #6: clearer warning if wrong Python version.

### PR DESCRIPTION
**Pull request by ablondin**
If the Python version is <= 2, a warning is displayed about how to
correct the problem. This solves issue #6.

### CODE REVIEW

**Comment by ablondin** [comment]
@sim590: please review.

File: bin/process.py [hunk 1]
```diff
@@ -1,5 +1,14 @@
 #!/usr/bin/env python
 # coding=utf-8
+
+from __future__ import print_function
+import sys
+
+# Check for version
+if sys.version_info[0] <= 2:
+    print('Only works with Python 3. Please either change the first line of /usr/bin/yauml or make `python` points toward Python 3.')
```
Comment (sim590) [inline bin/process.py]: I don't think that the user should be changing the first line of the script as explained in my other comment.

If the user's `python` would eventually be pointing to something different from the OS maintainer's choice (changing the symbolic link), wouldn't it eventually lead to breaking the system's python package (for e.g. after a package upgrade)?

File: bin/process.py [hunk 2]
```diff
@@ -1,5 +1,14 @@
 #!/usr/bin/env python
```
Comment (sim590) [inline bin/process.py]: On my system, Python executable files under `/usr/bin` look like this:
```sh
$ ls -l /usr/bin | grep python
...
python -> python3
python2 -> python2.7
python2.7
python3 -> python3.4
python3.4
...
```
I think that it does on all systems. Depending on your system maintainers choice, python may point to `python2` or `python3` which in turn points to the specific python version executable for instance, `python2->python2.7`. 

That's why I think that if the program runs on python 3, this line should be:
```python
#!/usr/bin/env python3
```
Then calling the script with it's path would always lead to calling `python3`. Do you think there's any drawback doing this?

**Comment by ablondin** [comment]
@sim590: I agree with your objection, my solution could be cleaner. But `python3` does not exist by default in my case (MacOS 10.10.3 Yosemite):

     [master] blondin_al [~]
     $ ls /usr/bin | grep python
    python
    python-config
    python2.6
    python2.6-config
    python2.7
    python2.7-config
    pythonw
    pythonw2.6
    pythonw2.7

     [master] blondin_al [~]
     $ python3
    -bash: python3: command not found

Therefore, your solution of changing the line to `python3` is correct, but the way to correct it if it does not work would be to suggest to the user to add a symbolic link `python3` pointing toward `python3.X` (in my case, I use Macports):

    sudo ln -s /opt/local/bin/python3.4 /usr/bin/python3

By the way, it seems to be a problem about MacPorts only (see https://trac.macports.org/ticket/34326) 

I'll update the pull request accordingly.

**Comment by ablondin** [comment]
@sim590: Updated. Please review.

**Comment by sim590** [comment]
Yes! As referenced on the link you provided, [PEP-0394](https://www.python.org/dev/peps/pep-0394/) specifies the symbolic link requirements. So this final solution is the right one.

**Comment by sim590** [comment]
Before merging this change, can you make a new PR for the same changes, but merging into the `dev` branch instead of master?

**Comment by ablondin** [comment]
Closed since this is solved by pull request #10, which targets the dev branch instead of master.

**Pull request closed without merging**